apis/Google.Longrunning/Google.Longrunning/OperationFailedException.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Longrunning
+{
+    /// <summary>
+    /// An exception to indicate that a long-running operation failed.
+    /// </summary>
+    public class OperationFailedException : Exception
+    {
+        /// <summary>
+        /// The operation message containing the original error.
+        /// </summary>
+        public Operation Operation { get; }
```

Comment:
Should this be called `Proto` to reflect `Operation<T>`'s property name and the constructor's parameter name? If not, maybe the parameter name should be changed to operation.